.github/workflows/test.yml

```diff
@@ -64,13 +64,16 @@ jobs:
           it:compile
 
       - name: Upload Test Report
-        uses: mikepenz/action-junit-report@v1
+        uses: mikepenz/action-junit-report@v2.0.3
         with:
           report_paths: '**/target/test-reports/TEST-*.xml'
+          check_name: Test Report ${{ github.event_name }}
           github_token: ${{ secrets.GITHUB_TOKEN }}
 
       - uses: codecov/codecov-action@v1
 
       - name: Fail if tests failed
         if: ${{ steps.tests.outcome != 'success' }}
-        run: echo "::set-failed ::Tests failed!"
+        run: |
+          echo ::error ::Tests failed! See "Test Report" check or previous output for details.
+          exit 1
```

Comment:
1.  GHA documentation has a warning to not use double quotes:  https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#about-workflow-commands
2. For the life of me, I could not get `::set-failed` to work. Since `::set-failed` is a shortcut for `::error` and `exit 1` (https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#using-workflow-commands-to-access-toolkit-functions),  I just made it explicit.